### ISSUE TITLE
General NOT WORKING etc. thread...

### ISSUE THREAD

**Issue by LRX22G** [pylerSM/YouTubeBackgroundPlayback#336]
Can you upload anywhere else?

**Comment by lobezno223** [comment]
+1

**Comment by th3an7** [comment]
I think the only way to share it is through some personal cloud storage...

In case someone needs an apk of a module... [I'll leave it here](https://drive.google.com/file/d/0B8j7vCsf6RXcUGdQSzdwV1ZWNzA/view?usp=sharing)

**Comment by LodanZark** [comment]
There is no build bots?

**Comment by th3an7** [comment]
Nope... Why would you need that if every update for YouTube app is made on
GitHub and then sent straight to your phone?

There are no changes in app source code beside hooks which are downloaded
remotely.

**Comment by sigv** [comment]
Anybody know what actually happened and why it is no longer on the Xposed Repo?

**Comment by iHelp101** [comment]
One of the other developers can choose to explain the matter further, but I will let you know it was not a developer that removed it. Overall, I think you can guess why the module was removed.

**Comment by LodanZark** [comment]
Because allow to adquire a paid feature for free...

**Comment by lobezno223** [comment]
I think the main question current users of this module have is if we can continue updating yt app automatically with no worries...

**Comment by thrdroom** [comment]
@iHelp101  -.-  why the mystification of any real information? At all the DEVs: Just tell the people all the reasons upfront! Whats the matter of being 100% transparent, god damn?!

**Comment by th3an7** [comment]
@thrdroom In a nutshell - rovo (Xposed owner) got an email from Google that he has a module on his repo that makes YouTube app playing in a background for free + it's messes up with app code which is against YouTube ToS. He had to remove the module from official repo...

I guess you can figure out the rest...

**Comment by thrdroom** [comment]
@th3an7 Thanks a lot for the clarification. Well, thats sad to hear :( so where to go now? Any chance of getting this project keep going without the need of xposed? Any ideas of how to avoid getting punished by google the same way?

**Comment by iHelp101** [comment]
@thrdroom In terms of why I was "mysterious" about the situation it is because takedown request can be confidential in some cases. Especially since it was sent to another party and relayed to us. I have mentioned to the other developers that Google gave a slap on the wrist with this. If they want to they will just takedown the GitHub as well in the future. In the end Google doesn't like what the module does. 

You can go the smali hacking route, but distributing an APK that has been smali hacked/modified would be distributing all of Google's codes and copyrighten images, which is going to get one in even more trouble compared to the slap on the wrist in this case.

**Comment by th3an7** [comment]
You know... The sources are still here - everyone can build an apk for
themselves. The main problem is, if somehow Google will get to know that we
are still updating the sources, they could force to delete them too -
that's what we want to avoid for now.

I think users should learn to update and build the module themselves...
It's really not that hard -  you just need to know what to search for :)
It takes me max. 10 minutes to update all the sources (original & fork + 1
branch)

**Comment by th3an7** [comment]
[There is even a tutorial I've made some time ago... just please, mute the video and turn on CC for English subs :P](https://youtu.be/_rgOsdq0-Lo)

**Comment by TheEmpressMiaw** [comment]
For those who can't build the apk for themselves, Me along with ZaneZam, Razerman and Master_T is building and regularly updating YouTube app with Background Playback and adaway module on xda.

https://forum.xda-developers.com/android/apps-games/app-iytbp-injected-youtube-background-t3560900

We provide
1. Root version
2. Magisk version
3. Non root version
4. Red Black themed version
5. Grey Black themed version

You can even enable/disable background playback via settings

**Comment by Lightfire228** [comment]
So, the next question is, will this repo be updated after each release?  I just read the tutorial on Xposed modules by rovo, so I don't think it would be too hard to do this on my own with this repo as a base.

Heck, I'm still stuck on YT 12.15.55 for YT adblock

**Comment by th3an7** [comment]
Maybe I will update it when new version is released... but won't promise
anything :)

**Comment by gokulraj3** [comment]
I would want to know that, can you update the module for us in the future.?

**Comment by gokulraj3** [comment]
And is there anyway that you can distribute it to us.? Without Google catching in.  Background play is my favorite module.  For which I installed Xposed..

**Comment by th3an7** [comment]
As I said before - if they won't change much, I'll try to update it.
About distribution - personally, I don't know... I guess there could be
"rain" from my personal cloud (if you know what I mean :P)

**Comment by Lightfire228** [comment]
Since we're all on github, I assume we're all devs.  We can just compile the apk from source using Android Studio.  That's what I did to my newly rooted Kindle Fire.  I feel like that would be safer, since Google can't claim the apk is being distributed.  (just my opinion though.  I've not had to deal with legal stuff like this before)

**Comment by terrorist96** [comment]
Is there something we can subscribe to in order to get notified of new updates for this module?

**Comment by Vednix** [comment]
https://infinityssh.ddns.net/ytbp/
Debug Apk + Release Apk + Source Gradle (AS)

**Comment by KimJiSung1** [comment]
@Vednix it doesn't work

**Comment by th3an7** [comment]
@KimJiSung1 uninstall the one you have now (if you have one ofc)

**Comment by KimJiSung1** [comment]
@th3an7 It worked, thanks

**Comment by vadinoguerrero** [comment]
Been using this module for a long time, and it being taken down is a tragedy, besides, thank you so much for it and for sharing it again

**Comment by cgarz** [comment]
This is really terrible. Youtube neglects to impliment background play for years to the point people were patching the apk and redistributing it. Then xposed made the whole process easier and without breach of copyright. Then years later youtube decides to copy this modules idea. But ONLY for extra money. And somehow this module is in the wrong?! and was told to be taken down?! Is there anywhere we can tell them just how dispicable they are? Or a petition to overturn this spiteful atrocity? I wish them all the worst in life.

**Comment by gokulraj3** [comment]
hey, can make this app for without root and without xposed. is it possible?

**Comment by th3an7** [comment]
@gokulraj3 I guess only by modifying YT app itself... but you won't be able to login to your account

**Comment by gokulraj3** [comment]
@th3an7  THanks for the reply.. i think you are correct. but having a app like that also will be cool too..

**Comment by gokulraj3** [comment]
The YouTube app got updated, please help me .. the module stopped working in in my phone after the new update.

**Comment by th3an7** [comment]
I've just updated the sources... It should work now...

**Comment by gokulraj3** [comment]
@th3an7  thanks so much..

**Comment by rmenessec** [comment]
Why isn't this posted in the [Releases section](https://github.com/pylerSM/YouTubeBackgroundPlayback/releases)? GitHub provides homepage hosting, wiki hosting, binary release storage...

**Comment by LodanZark** [comment]
@rmenessec probably to keep the updates more low profile to avoid getting the risk of this repo being removed from github...

**Comment by rmenessec** [comment]
@LodanZark Then it's probably a very bad idea to host builds via Google Drive, too.

**Comment by th3an7** [comment]
@rmenessec we're not changing .apk so much, only the code for supporting YT app... Module was removed form Xposed Repo, we don't want it to be removed from GitHub too.

Also, that Drive is my private cloud so... let's say it not "official" release :)

**Comment by LodanZark** [comment]
@rmenessec Not exactly, making the binary available in the project page notifies the followers of the repo, plus makes it easier to share the "official" download links which calls more attention than someone shares compiled files from the google drive personal links

**Comment by LodanZark** [comment]
th3an7 was more faster to reply than myself xD

**Comment by dakotagrvtt** [comment]
@rmenessec Maybe try Mega too?

_Sent from my Oneplus A0001 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by rmenessec** [comment]
@dakotagrvtt 
> @rmenessec Maybe try Mega too?

Sorry, I don't understand. This doesn't address the points I made. See my edited comment above.

If Google really have a problem with this project, changing a hosting provider won't make much difference. Changing how notifications are provided won't do much, either. If YTBP is using parts of Google's code—I assume this is **NOT** the case—it needs to be rewritten not to.

If Google doesn't have any real legal basis for objecting to YTBP... which I suspect is the case... it shouldn't matter where the project is hosted.

**Comment by th3an7** [comment]
@rmenessec the problem is, module it self does not have G's YT app code - it connects to it through Xposed and thus modifying it allowing to get paid content for free - background playback is a part of YT RED subscription.

Nobody would like to have is paid content be free using hacks, right? That's why they issued take down from Xposed repo with all that breaking ToS stuff.

**Comment by rmenessec** [comment]
@th3an7,

> Nobody would like to have is paid content be free using hacks, right? That's why they issued take down from Xposed repo with all that breaking ToS stuff.

I understand that reason, yes. But that's not a **legal basis** for threatening you or other developers. If the threat was over an EULA, I can't advise you on that. (IANAL) I can state with reasonable certainty, though, that they can't issue a (valid) DMCA takedown against the GitHub project if you're not infringing a copyright held by Google. You're not copying or otherwise redistributing their software, so it looks to me like there's no possibility of infringement.

If you want to get rid of the showdown over the EULA, though, and f the threat is specifically against the YouTube app's EULA and not the service, then you may want to change your approach altogether. Instead of modifying the app—I think Google are very wrong to worry about whether you're changing a temporary copy in memory, but that's me—you may want to wander over to the NewPipe project and ask what you can do to help them reach a 1.0 release.

... That, or work on your own project using entirely third-party software, instead of altering the behavior of the YouTube app. There's not a lot they can complain about if you do work on software that's 100% yours... or at least 100% not **theirs**. 😉

**Comment by terrorist96** [comment]
@rmenessec What you say is probably correct. However, just because what you are doing is not illegal doesn't mean you can't be arrested, prosecuted, fined and imprisoned for not breaking the law. Look at Megaupload as a perfect example. On the surface, it was no different than Google Drive or Dropbox. But they got shut down, raided, arrested, assets forfeited, and been fighting a legal war for over five years now, and still haven't been convicted of anything yet. And I doubt the people maintaining this project have the resources of Kim Dotcom to put up a fight where they may be in the legal right. Invalid DMCA notices are sent all the time and are complied with, and the consequences for invalid DMCA notices are rarely enforced.

**Comment by cgarz** [comment]
I see. I didn't think YTBP was at fault but the discussion on legality here has further convinced me. They really are dispicable then. I've seen blatant abuses like this in other places where organizations try their luck. Large widespread public outcry can help. We really should petition or something. Maybe even ask EFF or similar if you have a good case against them that you could hit them with.

**Comment by Meneerdaniel** [comment]
Hey, I noticed it wasn't working properly anymore. What do I need to do to get it working again? The version that is still on my phone(Xposed method) is 2.0.4. I'm not a developer so I have no idea what to do know. Help would be greatly appreciated!

**Comment by th3an7** [comment]
Which version of YT app do you have? If it's today's update, it should work
again. I've already updated it. Try force closing YT app or reboot your
phone

**Comment by Meneerdaniel** [comment]
O yeah you're right. I didn't do a reboot. Thanks! Great tool/app!

**Comment by cherryleafroad** [comment]
I didn't see/know that @th3an7 updated built his module regularly. So I built my own copy. I also changed the versioning to include the specific git commit it was built from (very useful imho). Just leaving this here anyways.

https://www.mediafire.com/file/b2fi35h1alwbs61/YouTubeBackgroundPlayback-master-release.apk

**Comment by xkillerz** [comment]
@th3an7 You can build apk and upload it to github as "release". Try it!

**Comment by th3an7** [comment]
I know, but If I do this, they could also find it and try to take down this
repo - which we don't want :P

**Comment by xkillerz** [comment]
Oh, I didn't realize this.🤔

-Zhen Xu

2017-08-31 11:34 GMT-04:00 Łukasz Ordon <notifications@github.com>:

> I know, but If I do this, they could also find it and try to take down this
> repo - which we don't want :P
>

**Comment by Zralle** [comment]
@th3an7 Not working after last yt update (12.34.54), yes I rebooted device twice :D 
If you need any additional information just say.

**Comment by fragtion** [comment]
Wow, pathetic. I've noticed YouTube and Google in general are starting to bow to the pressure of their corporate sponsors in a number of ways that are starting to piss me as the user off greatly. I'm reaching the point where I'm starting to look for alternative for the likes of my daily Searching, E-mail, and multimedia requirements and will pursue this route more aggressively if Google keeps up this bullshit

**Comment by Zralle** [comment]
@fragtion I believe that th3an7 will update module soon. Be patiant new update just came out few hours ago.

**Comment by th3an7** [comment]
Sources updated - try FC app/rebooting phone :P

**Comment by cherryleafroad** [comment]
[Newest 5c55f3d build, support YouTube 12.34.54](https://www.mediafire.com/file/2f22u5py657v46z/YouTubeBackgroundPlayback-master-release-5c55f3d.apk)

**Comment by th3an7** [comment]
You don't need to build new APK everytime hooks are getting updated. They
are automatically downloaded from this repo.

Building new apks is needed only when .java file gets changed :P

**Comment by cherryleafroad** [comment]
Well, I see :P That's a pretty nifty feature! 💯 While I'm on this track of thinking about xposed and hooks, I read about how some of the developers were able to figure out the location of certain hooked methods dynamically by checking out places in the code that doesn't change with proguard, that way they don't have to find the right methods every time (the app automatically knows, just need to force close and restart it after an update). What do you think about that? @th3an7

**Comment by th3an7** [comment]
Well... This repo is not mine - I am just updating the hooks :P
Finding them takes me max. 2 mins...

Also, in this today's update, they've added new .dex file which isn't big
as the other two (~160KB) but it seems like they may be preparing for
another big change in YT app code...

Let's wait couple of updates and see how it goes :P

**Comment by davedavis** [comment]
@cherryleafroad Thank you so much! Works great!

Happy to donate to future updates if needed.

**Comment by myself379** [comment]
I'm hunting down for 12.36.56 but it looks like it has 3 classes from 12.36.55

They renamed "background_mode", "background_audio_policy" and "acwa" in 12.36.56

Where do you search for the names? I have no clue on how to hunt it down now

**Comment by gokulraj3** [comment]
Hi Guys, I am Crying Now. I have changed my device. I don't know whether I should say this here, but sorry if I posted this in the wrong place. I am using the Lenovo k8 note since a month now. It has not received a custom TWRP Recovery or anything. I am Also not able to Root the Device. If anyone in here is a developer please help me out in this problem. So many Lenovo k8 note users are also waiting for the rooting file to come. If anyone here is a developer help me.😭😭😭

**Comment by areco23** [comment]
Hey mate they all ready Cheney YouTube so the background play is not working again. Any plans to fix it?

**Comment by meshal707** [comment]
Update 37.59 Where do I find the update file and thank you for the above

**Comment by th3an7** [comment]
@meshal707 you mean new module .apk? You don't need one - reboot or force close YT app...

If sources are updated for latest YT version, then it will work automatically - no need of installing any updates of module :)

**Comment by meshal707** [comment]
First I would like to thank you I was using kitkat 4.4.2, I updated the
application without problems and I was updated ready to Lineage Os and then
the shock was when I did not find it in the library After searching, I
found her, but I am afraid of modernization and her loss again. For that, I
asked, I do not want to risk it at all now I'm afraid of updating youtube
and I can not find the app then I will cry . I respect you and my
appreciation .
Note: I use Google translation and I do not know about spelling is it true
or not .
Sorry

بتاريخ ٠٦‏/١٠‏/٢٠١٧ ٥:٢٠ ص، كتب "Łukasz Ordon" <notifications@github.com>:

**Comment by meshal707** [comment]
I have updated to 12.37.59 and it works normally thank you

بتاريخ ٠٦‏/١٠‏/٢٠١٧ ٦:١٥ ص، كتب "مشعل العاصمي" <m.92.alasiri@gmail.com>:

**Comment by SandPox** [comment]
APK of 43d6b5902ce800011e5e2d01ff9e39cdf6db368f (7 Oct 2017)

https://drive.google.com/open?id=0B0MI5d7Kg-kyWHRkWUQyZFhyYzg

**Comment by KimJiSung1** [comment]
@SandPox 

> https://drive.google.com/open?id=0B0MI5d7Kg-kyWHRkWUQyZFhyYzg

It appears to be corrupted

**Comment by meshal707** [comment]
I update it to YT.59 and it's work smoothly thank you for the help and I appreciate that

**Comment by SandPox** [comment]
Hmm.. I don't know, probably something happens while reuploading

Could you test this: https://build.sandpox.org/YouTubeBackgroundPlayback-release-unsigned.apk

If it's working fine then pls reupload it since it only save temporary on there

**Comment by KimJiSung1** [comment]
# ![screenshot_20171007-225518](https://user-images.githubusercontent.com/30732876/31311727-a6882bd6-abb2-11e7-80e0-f738f83d7983.png)
@SandPox

**Comment by SandPox** [comment]
@KimJiSung1 
Ah, it install failed with reason INSTALL_PARSE_FAILED_NO_CERTIFICATES because it's unsigned, the debug build seems to be install fine
https://build.sandpox.org/YouTubeBackgroundPlayback-debug.apk

**Comment by KimJiSung1** [comment]
@SandPox thank you, it works <3

**Comment by Kocal** [comment]
@SandPox thanks :)

**Comment by cherryleafroad** [comment]
@SandPox The app automatically downloads the json hooks file, so no need to rebuild the apk every update. :) Also, you can try [https://developer.android.com/studio/publish/app-signing.html#sign-apk](self signing) so that you can give a release build (not debug)

**Comment by VladMano** [comment]
Hello, I know that this is random but does this module work with the official xposed nougat that rovo released today? I have not followed this module for a long time since I was on nougat without xposed and I kinda lost track of everything.

**Comment by cherryleafroad** [comment]
@VladMano there's no reason it shouldn't :)

**Comment by VladMano** [comment]
@cherryleafroad I installed SandPox's version and it works perfectly on yt 12.37.59, Feels good to be back, Thanks!

**Comment by Kocal** [comment]
@VladMano yes it works with the official Xposed Nougat, I'm actually using it!

**Comment by borisov87** [comment]
After youtube is updated to 12.39.60, background stop work. I try restart phone withou effect.

**Comment by someone755** [comment]
@borisov87 Well no shit, even yesterday's update doesn't cover that.

**Comment by th3an7** [comment]
@borisov87 @someone755 - did it work yesterday? There was no .60 YT apk update yet so...

I'll fix it later today...

**Comment by borisov87** [comment]
Yes yesterday i listen music :)

**Comment by th3an7** [comment]
Should work again - reboot or force close YT app :P

**Comment by JohanSF** [comment]
Do we need a new apk?

**Comment by th3an7** [comment]
No. If you'll need new module apk, you'll know :)

**Comment by cherryleafroad** [comment]
@JohanSF No, the hooks file is downloaded automatically, so it updates itself dynamically. No need for a new apk

**Comment by Hrybok** [comment]
If I'm using 12.23.60, which version should I download in order for it to work? 
Would the latest version work without a problem?

**Comment by th3an7** [comment]
It doesn't matter - all hooks are downloaded from the repo so if it's
updated, it should work for whatever module version you're using :)

**Comment by Hrybok** [comment]
Amazing, thank you so much for your excellent work!

**Comment by borisov87** [comment]
On 12.43.52 stop work

**Comment by th3an7** [comment]
@borisov87 can you test if it works now?

**Comment by borisov87** [comment]
Tnx now all is ok

**Comment by lobezno223** [comment]
Hello,

module not working for latest YT update 12.44.52, under these device specs:

OS - Kitkat 4.4.2
Display - 320dpi
CPU Arch. - ARMv7 Processor rev3 (v7l)
armeabi-v7a (Kernel armv7l)

**Comment by th3an7** [comment]
There is no YT apk for 44.52 yet... upload it somewhere so I can update the
module or wait a bit longer till it'll be uploaded on apkmirror :P

**Comment by lobezno223** [comment]
OK, no problem, I'll wait :) my comment was just to report the issue.
I don't have the apk to upload. I updated the app directly from the play store and then the module doesn't work.

**Comment by lobezno223** [comment]
anyway I found the 12.44.52 apk here:

https://download.dlapk.org/?apkid=com.google.android.youtube&ver=12.44.52&server=amd123

But don't know if it's better to wait for Apkmirror's  liability and variety of versions to work with.

**Comment by th3an7** [comment]
@lobezno223 can you test if it's working again? Force close YT or restart the device :)

**Comment by lobezno223** [comment]
yes it works fine now. thanks for the good work, appreciate it

**Comment by innuendofloyd** [comment]
Doesn't work anymore on yt 12.44.53 :(

**Comment by th3an7** [comment]
@innuendofloyd there is no 44.53 on apkmirror... upload YT .apk somewhere or wait till it's available :)

**Comment by innuendofloyd** [comment]
Lol meantime I was uploading the apk :D

**Comment by innuendofloyd** [comment]
I Confirm it works!

**Comment by ktl8** [comment]
It was working until 12.45.54. Or do I have to do something to make it work? Restarting it didn't work

**Comment by th3an7** [comment]
@ktl8 should work now - force close YT app or reboot your device :)

**Comment by blackman91** [comment]
Could anyone share the latest YouTube background play apk with me please? Pm a link please or post it here.

**Comment by blackman91** [comment]
Not working as of today with latest YouTube. Please fix.

**Comment by th3an7** [comment]
@blackman91 it's updated already... Check your logcat/Xposed logs

**Comment by dimateos** [comment]
Working perfectly! thanks for the updates

**Comment by KimJiSung1** [comment]
Doesn't work with 12.**44.53**

**Comment by KimJiSung1** [comment]
Doesnt work with 12.45.54 also?

**Comment by dimateos** [comment]
@KimJiSung1 Works for me with 12.44.53 but I cannot update to 12.45.54 yet sry

**Comment by KimJiSung1** [comment]
**@dimateos**  It did work for me first but then it stopped working. I tried uninstalling and installing it again.

**Comment by panyuhang93** [comment]
it's me for 12.47.57:goo.gl/LF1hQa

**Comment by KimJiSung1** [comment]
Hey can somebody help. I am getting  "The The package conflicts with an existing package by the same name." Error

**Comment by blackman91** [comment]
Uninstall the one you have already installed and install the new one.

**Comment by gitdef** [comment]
Topic also here
[https://tinyurl.com/yb3r77vh](url)

**Comment by Zralle** [comment]
Not working on Xposed 89 update.

**Comment by shrx** [comment]
Not working.
Samsung SM-A320FL
Android 7.0 (API 24)
Xposed Framework 89.0
Youtube apk version 12.47.58

**Comment by misterpyrrhuloxia** [comment]
Here is the latest version of the YouTube app:

https://drive.google.com/file/d/1DE1MjU_t_75tqGr3EBGlCnq9C6Qar-Tw

**Comment by th3an7** [comment]
@misterpyrrhuloxia thanks - will update the module in a second...

One thing - is is for android 4.1+ or 5.0+ or what Android version are you on right now?

**Comment by misterpyrrhuloxia** [comment]
I am on 7.1.2.

**Comment by misterpyrrhuloxia** [comment]
@th3an7 are you planning on making a new tutorial video to replace the obsolete method video? I would love to be able to contribute by following your directions each time my Youtube app updates. I seem to get each new version before it's even added to apkmirror. So I could be useful because of how soon I get the updates.

**Comment by panyuhang93** [comment]
For me, I think the update is fast enough by appmirror,How did you do so fast update

**Comment by misterpyrrhuloxia** [comment]
I don't know, tbh. ¯\_(ツ)_/¯    It just happens, usually.

**Comment by panyuhang93** [comment]
You are so lucky

**Comment by th3an7** [comment]
@misterpyrrhuloxia the video is here - https://youtu.be/OG8fvxnZnSo ...
sorry for my shit commentary - just mute it and turn on CC for english description xD

PS: I've used IDA which isn't free but I think you'll find it somewhere ;)

**Comment by misterpyrrhuloxia** [comment]
Thanks!  :)

**Comment by newstart** [comment]
@th3an7 if your cloud storage is feeling a little stuffy and needs to let it out, I would not mind a little rain in my direction.

**Comment by th3an7** [comment]
@newstart 
- [version for Android 5.0+](https://drive.google.com/open?id=0B8j7vCsf6RXcUGdQSzdwV1ZWNzA)
- [version for Android 4.1+](https://drive.google.com/file/d/1TgfEVRczbhezLcJO6qkofQxcZ5PH81yX/view?usp=sharing)

So choose wisely :P

**Comment by brnl** [comment]
Thanks @th3an7 for your great work. This module is one of my absolute favorites!

**Comment by newstart** [comment]
@th3an7 wtf.... This enables PiP? Shit this is even better than normal background play! You freaking rock dude!!! Thanks for the rain!!!!

**Comment by misterpyrrhuloxia** [comment]
What is PiP?

**Comment by th3an7** [comment]
@misterpyrrhuloxia Picture in Picture... Android 8.0+ feature...

It's probably side effect of turning on background playback :P

**Comment by panyuhang93** [comment]
i find pip too, so great

**Comment by fragtion** [comment]
To the guys with PiP working, are you already on Android 8.x ? Don't want to waste my time replacing Magisk YTBP module with this APK if it won't work on Nougat (7.x)

**Comment by newstart** [comment]
@fragtion, I am on Oreo. I think it is only oreo specific.

**Comment by misterpyrrhuloxia** [comment]
Will the new hook update method work with the evaluation version of IDA 6.8? Or do I have to have IDA 6.8 Pro?

**Comment by misterpyrrhuloxia** [comment]
Btw, here's the latest Youtube App version 13.03.58:

https://drive.google.com/file/d/1xx2p9Cenm-B0EFyoVDfwJt3S6iNyRntZ

**Comment by brnl** [comment]
@misterpyrrhuloxia 

I would strongly recommend getting the official APK from https://www.apkmirror.com/apk/google-inc/youtube/youtube-13-03-58-release/ instead of some strangers Google Drive. Unless a verifiable hash is provided, I wouldn't trust it.

**Comment by misterpyrrhuloxia** [comment]
Ok. That's actually the updated apk that I uploaded to my google drive from
my phone right after I updated the Youtube app several hours ago.

**Comment by brnl** [comment]
@misterpyrrhuloxia That's ok. I am not accusing you from anything or saying that I don't trust _**you**_, but it is still an unverified and unknown source. So in general it's good practice to grab an APK from a known source if that's available. Not to say that I _know_ apkmirror.com that well, but they seem reasonably trusted by the community ;-)

**Comment by misterpyrrhuloxia** [comment]
That makes sense.

**Comment by th3an7** [comment]
@misterpyrrhuloxia I guess as long as you have some decompiler that have some hex view, it should be possible...

I've used that version because it was first Google result :D 



PS: Update for 13.03.58 for Android 5.0+ should be up in a 30 mins max
                                       for Android 4.1+ - no YT .apk available yet

**Comment by lobezno223** [comment]
Hello,

I decided to not update yt since 12.42.59, which was the last working
version for the previous ytbp apk, until now.

So, can I finally download the latest versions of YT for android 4.4.2, and
this versions of ytbp will work?

many thanks to all, the rebirth of this thread is a great surprise to me.

2018-01-30 16:40 GMT+01:00 Łukasz Ordon <notifications@github.com>:

**Comment by th3an7** [comment]
>So, can I finally download the latest versions of YT for android 4.4.2, and
this versions of ytbp will work?

Depends which version is the lastest one - if it's 13.03.58, it is not yet updated (I guess it'll be live tomorrow) but for the older YT version it should be working - you just need KK module version

**Comment by lobezno223** [comment]
It is 13.01.53, actually.

What is KK module version?

**Comment by th3an7** [comment]
It's the version made for Android 4.1+ - it's [here](https://github.com/pylerSM/YouTubeBackgroundPlayback/issues/336#issuecomment-361030649) if you need it

**Comment by misterpyrrhuloxia** [comment]
So I am not a developer at all and don't really know what I'm doing. Is
there another decompiler you could recommend that's free that will be
capable of the steps in your updated video?

**Comment by th3an7** [comment]
@misterpyrrhuloxia so...

I've tried several free disassemblers and decompliers and none of them were capable of what IDA can do...
Not sure about that but you could also try decompiling .dex to .jar and read all classes by yourself - but that's stupid cause they are changing their places - finding 2 first classes is relatively easy (you just search for string)...
The third one is tough tho - there is no strings to search for so old java reading method is useless - that's why disassembler with hex viewer is a must - instead of reading 2000 classes, you just have 50 or even less (if you know what to search for).


TL;DR - nope... just get some "modified" IDA version, it's the fastest way to do it :P

**Comment by blackman91** [comment]
Guys this is a great alternative for YouTube background play and no ads, it doesn't require xposed:
https://forum.xda-developers.com/android/apps-games/app-iytbp-injected-youtube-background-t3560900

**Comment by kevincaradant** [comment]
Hi everyone, 

@th3an7,
I trie3d your version but everytime after one video with screen off, the next track  isn't read like the background service was killed so I unlock the phone, open YouTube and I get "audio available" but it's freeze until I back and select again a video, any idea?

![s80204-15265129](https://user-images.githubusercontent.com/6093175/35778590-63f177de-09c0-11e8-936a-c6ce49300952.jpg)

**Comment by th3an7** [comment]
If you're using some RAM cleaners app (greenify etc.), turn them off...

Also, try disabling battery optimization in Android settings (if your OS
has it)

**Comment by kevincaradant** [comment]
Hello, I already check them and allow to running in background.
If I enable the screen even on the lock screen, the videos run fine. 

That's really weird but am I the onlynone to have that?

**Comment by besentv** [comment]
Do you accept translations? Also lint seems to cause some problems with the missing strings in some translations.

**Comment by th3an7** [comment]
@kevincaradant then in may be your device - if it has low RAM, OS can kill it itself - it may be YT app fault, not the module

@besentv since the module does not provide any UI, the only string that is translatable is app name - so I don't think they are that important... but if you want, you can always create pull request :)

**Comment by kevincaradant** [comment]
My device maybe but I'm really surprising that this problem seems to be appeared really recently, not at the begin, I have 3G of ram with 2G used.

My workaround is to use a screen on application. Not really nice for the drain battery but at least it works

**Comment by th3an7** [comment]
@besentv I've squashed your pull request to just one commit if that's OK - thanks for contributing :)

@kevincaradant if it's recent, then it may be definitely YT app fault - the module code hasn't been changed since like 8 months now - the only change there was, were hooks which are updated here on github and send directly to the module through internet...
Some users reported it long time ago and since then, there was no other reports - so yeah, it may be YT recent updates...

**Comment by besentv** [comment]
@th3an7 That's perfect. Don't want others to see how I messed up. xD

**Comment by sutobe** [comment]
Hey th3an7, 

thanks for your efforts.  Never tried this before, also wondered where it did go xD

Anyways, it is working on my HTC 10 with Nougat 7.0 (don't want to upgrade on Oreo because meh, I have absolutely no need and there are modules and apps on my phone not working with O).

There is just one thing that is a little annoying and I am wondering if this could be solved somehow, when I open YT, play a video, jump back to the main screen with the middle navigation button it of course shows me the YT symbol in the notification bar which lets me go back to the video or skip clips in a playlist.

So far so good, but when I go back into the app or close Youtube for good that notification symbol stays in the notification bar just showing the greyed out control buttons, I have to manually close it.

Maybe this can be fixed somehow so it will go away when you go back into the youtube app or close it?

Cheers

**Comment by th3an7** [comment]
The module doesn't create the notification - it only shows it from the
app... If there is something wrong with it - it's YT app fault :P

**Comment by sutobe** [comment]
oh ok was just a thought. anyways it is working well.
Just read something about PiP, how do I get this to work? Never tried it before xD

**Comment by th3an7** [comment]
Picture-in-picture is Android Oreo exclusive - it's an OS feature... It
works automatically on 8.0+ - just minimize app and video will play inside
small window on top of everything else

**Comment by man0usos** [comment]
Can someone post the updated apks please?

**Comment by th3an7** [comment]
@man0usos they haven't changed since they were posted earlier... And they probably won't in the future (at least for now)

But they are here https://github.com/pylerSM/YouTubeBackgroundPlayback/issues/336#comment-361030649

Also, if the module is working fine, you don't need to install anything from there - all is updated here and pushed to your phone directly when you open YouTube app

**Comment by panxuelei** [comment]
can't use in the latest version of YouTube
最新版的YouTube 不能用

**Comment by panyuhang93** [comment]
@panxuelei 现在支持到13.11.57

**Comment by th3an7** [comment]
Anyone interested, please check #357

**Comment by th3an7** [comment]
Closing this as it starts to lag and opens for almost 10 secs...

Moving general reporting issues to #357 i guess
